edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/CloudConnection.cs

```diff
@@ -187,7 +187,7 @@ async Task<IClient> ConnectToIoTHub(IClientCredentials newCredentials)
             {
                 client.SetProductInfo(newCredentials.ProductInfo);
             }
-
+            await client.OpenAsync();
```

Comment:
What happens during an `OpenAsync`?

Reply:
It basically opens a connection to IoThub.. that way, we only get a DeviceClient/ModuleClient if it can reach IoTHub, otherwise, every time we retry, it tries to open a new DeviceClient..

Reply:
OK. How do the subscriptions get synced? Are they synced after this open call succeeds?

Reply:
Yes, that is correct. They are synced on the CloudConnection established event, so yes, after this Open call succeeds...

Reply:
Got it.